File: Capitulo_01/01_Getting_Started:_The_Easy_Problems/03_Medium/10424_:::_Love_Calculator/Main.cpp
using namespace std;
#include <cstdio>
#include <iostream>
#include <string>
#include <sstream>
#include <algorithm>

string parseInt(int number)
{
  stringstream ss;
  ss << number;
  return ss.str();
}

int main()
{
	string s1, s2;
	int sum1, sum2, i;
	double rta;

	while(getline(cin, s1))
	{
		getline(cin, s2);

		sum1=sum2=0;
		for(i=0; i<s1.length(); i++)
		{
			if(isalpha(s1[i]))
			{
				s1[i]=tolower(s1[i]);
				sum1+=(s1[i]-'a'+1);
			}
		}

		for(i=0; i<s2.length(); i++)
		{
			if(isalpha(s2[i]))
			{
				s2[i]=tolower(s2[i]);
				sum2+=(s2[i]-'a'+1);
			}
		}

		s1=parseInt(sum1);
		while(s1.length()!=1)
		{
			sum1=0;
			for(i=0; i<s1.length(); i++) sum1+=(s1[i]-'0');
			s1=parseInt(sum1);
		}

		s2=parseInt(sum2);
		while(s2.length()!=1)
		{
			sum2=0;
			for(i=0; i<s2.length(); i++) sum2+=(s2[i]-'0');
			s2=parseInt(sum2);
		}

		rta=(double)min(sum1, sum2)*100.0/(double)max(sum1, sum2);
		printf("%.2f %\n", rta);
	}

	return 0;
}

File: Capitulo_01/01_Getting_Started:_The_Easy_Problems/03_Medium/10424_:::_Love_Calculator/Main.java
import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.text.DecimalFormat;

public class Main
{
	public static void main(String[] args) throws IOException
	{
		BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
		//-----------------------------------------------------------
		DecimalFormat df=new DecimalFormat("0.00");
		char[] aux1, aux2;
		int sum1, sum2;
		double rta;

		for(String nombre1, cad1, cad2; (nombre1=br.readLine())!=null;)
		{
			sum1=sum2=0;
			aux1=nombre1.toCharArray();
			for(char c : aux1) if(Character.isLetter(c)) sum1+=(Character.toLowerCase(c)-'a'+1);

			aux2=br.readLine().toCharArray();
			for(char c : aux2) if(Character.isLetter(c)) sum2+=(Character.toLowerCase(c)-'a'+1);

			cad1=sum1+""; cad2=sum2+"";

			while(cad1.length()!=1)
			{
				sum1=0;
				aux1=cad1.toCharArray();
				for(char c : aux1) sum1+=(c-'0');
				cad1=sum1+"";
			}

			while(cad2.length()!=1)
			{
				sum2=0;
				aux2=cad2.toCharArray();
				for(char c : aux2) sum2+=(c-'0');
				cad2=sum2+"";
			}

			rta=Math.min(sum1, sum2)*100.0/Math.max(sum1, sum2);
			System.out.println(df.format(rta).replace(',','.')+" %");
		}		
		//-----------------------------------------------------------
		br.close();
	}
}
